pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -701,6 +775,22 @@ func (iqr *IQR) AsResult() (*structs.PipeSearchResponseOuter, error) {
 			log.Errorf("IQR.AsResult: error reading all columns: %v", err)
 			return nil, err
 		}
+		numOfRecords := 0
+		for col, values := range records {
+			if numOfRecords == 0 {
+				numOfRecords = len(values)
+			}
+			if numOfRecords != len(values) {
+				return nil, fmt.Errorf("IQR.AsResult: inconsistent number of rows for column %v, expected: %v, got: %v", col, numOfRecords, len(values))
+			}
+		}
```

Comment:
If we want to do this validation, I think moving it into `readAllColumnsWithRRCs()` would be better

Reply:
This was just an extra validation which I don't think is required since, we would validate this while appending IQRs anyways. Have removed it.